### PR TITLE
Implement optional Hyper 1 support in hyper-boring

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by Guiguiprim** [comment]
It would be really nice to have this as `hyper-boring` is the last reaming piece in my stack that doesn't yet support `hyper v1`.
Thank you very much for this !

File: boring-sys/Cargo.toml [hunk 1]
```diff
@@ -81,3 +81,6 @@ bindgen = { workspace = true }
 cmake = { workspace = true }
 fs_extra = { workspace = true }
 fslock = { workspace = true }
+
+[lints.rust]
+unexpected_cfgs = { level = "allow", check-cfg = ['cfg(const_fn)'] }
```
Comment (rushilmehra) [inline boring-sys/Cargo.toml:86]: I ended up just removing the `cfg(const_fn)`s, they seemed like a relic from when const wasn't stable

File: hyper-boring/src/lib.rs [hunk 2]
```diff
@@ -2,91 +2,27 @@
...
 use tokio_boring::SslStream;
-use tower_layer::Layer;
 
 mod cache;
-#[cfg(test)]
-mod test;
+/// Hyper 1 support.
+#[cfg(feature = "hyper1")]
+pub mod new;
+mod old;
```
Comment (inikulin) [inline hyper-boring/src/lib.rs:17]: can we name those modules `v0` and `v1` instead or something like that?

File: hyper-boring/tests/old.rs [hunk 3]
```diff
@@ -1,12 +1,12 @@
-use super::*;
```
Comment (rushilmehra) [inline hyper-boring/tests/old.rs:1]: We should probably name this `v0.rs` and name `hyper1.rs` `v1.rs` to be consistent with the implementation filse

File: hyper-boring/src/v1.rs [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::{io, net};
+use tokio::io::{AsyncRead, AsyncWrite};
+use tower_layer::Layer;
+use tower_service::Service;
+
+/// A Connector using OpenSSL to support `http` and `https` schemes.
```
Comment (rushilmehra) [inline hyper-boring/src/v1.rs:25]:
```suggestion
/// A Connector using BoringSSL to support `http` and `https` schemes.
```

File: hyper-boring/src/v0.rs [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+use std::future::Future;
+use std::net;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::{fmt, io};
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+use tower_layer::Layer;
+
+/// A Connector using OpenSSL to support `http` and `https` schemes.
```
Comment (rushilmehra) [inline hyper-boring/src/v0.rs:24]:
```suggestion
/// A Connector using BoringSSL to support `http` and `https` schemes.
```

**Review by rushilmehra — APPROVED** [review]

**Pull request merged**